portfolio/src/main/webapp/game_script.js

```diff
@@ -0,0 +1,199 @@
+
+/* multiple intruction boxes becaise
+   canvas does not support multi-line */
+var INSTR_UP;
+var INSTR_DOWN;
+var INSTR_MOUSE;
+
+function startGame() {
+  player = new component(30, 30, "red", 10, window.innerHeight/2);
+  score = new component("30px", "Consolas", "black", window.innerWidth - 200, 30, "text");
+
+  // create a component for each instruction box
+  INSTR_UP = new component("20px", "Consolas", "black", window.innerWidth - 250, window.innerHeight - 90, "text");
+  INSTR_DOWN = new component("20px", "Consolas", "black", window.innerWidth - 250, window.innerHeight - 65, "text");
+  INSTR_MOUSE = new component("20px", "Consolas", "black", window.innerWidth - 250, window.innerHeight - 40, "text");
+
+  // unicodes for down and up arrows
+  UPArrSymbol = "\u2B06";
+  DOWNArrSymbol = "\u2B07";
+
+  INSTR_UP.text = "w/" + UPArrSymbol + " > up";
+  INSTR_DOWN.text = "s/" + DOWNArrSymbol + " > down";
+  INSTR_MOUSE.text = "m > on/off mouse control";
+
+  gameArea.start();
+}
+
+/* Construct a component with specifiec width
+   height, color and coordinates where it is
+   places in the canvas */
+function component(width, height, color, x, y, type) {
+    this.type = type;
+    this.width = width;
+    this.height = height;
+    this.speedX = 0;
+    this.speedY = 0;
+    this.x = x;
+    this.y = y;
+    this.mouseControl = false;
+    this.update = function() {
+      ctx = gameArea.context;
+      if (this.type == "text") {
+        ctx.font = this.width + " " + this.height;
+        ctx.fillStyle = color;
+        ctx.fillText(this.text, this.x, this.y);
+      } else {
+        ctx.fillStyle = color;
+        ctx.fillRect(this.x, this.y, this.width, this.height);
+      }
+    }
+    this.newPos = function() {
+        this.x += this.speedX;
+        this.y += this.speedY;
+
+        if (gameArea.x && gameArea.y && gameArea.mouseControl) {
+            /* if mouse position exists and mouse control 
+               is enabled set player y coord to mouse y coord
+               - player only moves vertically*/
+            this.y = gameArea.y;
+        }
+    }
+    this.newSpeed = function() {
+        this.speedX = 0;
+        this.speedY = 0;
+
+        keyPressed = getKey(gameArea.key);
+        switch(keyPressed) {
+            case "UP":
+              this.speedY = -3;
+              break;
+            case "DOWN":
+              this.speedY = 3;
+              break;
+            default:
+              break;
+        } // switch
+    },
+    this.crashWith = function(otherObj) {
+      /* Get left, right, top, bottom for this object */
+      var l = this.x;
+      var r = this.x + (this.width);
+      var top = this.y;
+      var bttm = this.y + (this.height);
+
+      /* Get left, right, top, bottom for other object */
+      var otherL = otherObj.x;
+      var otherR = otherObj.x + (otherObj.width);
+      var otherTop = otherObj.y;
+      var otherBttm = otherObj.y + (otherObj.height);
+      var crash = true;
```

Comment:
You can simplify all of this by returning the boolean expression directly - you can negate it, because the expression is right now true when there is not a crash:
`return bttm >= otherTop && top <= otherBttm && r >= otherL && l <= otherR;`